File: src/main/java/io/github/oleksivio/telegram/bot/api/model/method/group/GetChat.java
package io.github.oleksivio.telegram.bot.api.model.method.group;

import io.github.oleksivio.telegram.bot.api.model.NetworkError;
import io.github.oleksivio.telegram.bot.api.model.objects.std.Chat;
import io.github.oleksivio.telegram.bot.core.controller.network.ActionNetworker;
import io.github.oleksivio.telegram.bot.core.model.responses.ChatResponse;
import io.github.oleksivio.telegram.bot.core.model.method.ChatAction;
import io.github.oleksivio.telegram.bot.core.model.responses.CommonResponse;

/**
 * @see <a href="https://core.telegram.org/bots/api#getchat">getChat</a>
 */
public class GetChat extends ChatAction<Chat> {
    private static final String METHOD = "getChat";

    public GetChat(ActionNetworker actionNetworker) {
        super(METHOD, actionNetworker);
    }

    @Override
    protected Class<? extends CommonResponse<Chat>> getResultWrapperClass() {
        return ChatResponse.class;
    }

    @Override
    public GetChat setChatId(Long chatId) {
        pSetChatId(chatId);
        return this;
    }

    @Override
    public GetChat setNetworkErrorListener(NetworkError onNetworkError) {
        pSetNetworkErrorListener(onNetworkError);
        return this;
    }
}

File: src/main/kotlin/io/github/oleksivio/telegram/bot/api/model/method/group/GetChat.kt
package io.github.oleksivio.telegram.bot.api.model.method.group

import com.fasterxml.jackson.annotation.JsonProperty
import io.github.oleksivio.telegram.bot.api.model.objects.std.Chat
import io.github.oleksivio.telegram.bot.core.model.ApiDict
import io.github.oleksivio.telegram.bot.core.model.ChatResponse
import io.github.oleksivio.telegram.bot.core.model.method.common.ChatAction

/**
 * @see [getChat](https://core.telegram.org/bots/api/#getchat)
 */
data class GetChat(
        /**
         * chat_id Integer or String Yes Unique identifier for the target chat or username of the target channel
         */
        @JsonProperty(ApiDict.CHAT_ID_KEY)
        override val chatId: Long
) : ChatAction<Chat>() {

    override val resultWrapperClass = ChatResponse::class

    @JsonProperty(ApiDict.METHOD_KEY)
    override val method = "getChat"

}
